tests/testthat/test-pstr_add_reductions.R

```diff
@@ -1,71 +1,82 @@
-  expect_false(hasName(pstr_toy_companies(), "year"))
-  expect_false(hasName(pstr_toy_companies(), "value"))
-  expect_false(hasName(pstr_toy_companies(), "reductions"))
+test_that("with pstr_toy_* companies outputs the expected columns", {
+  companies <- pstr_toy_companies()
+  ep_weo <- pstr_toy_ep_weo()
+  weo <- pstr_toy_weo_2022()
 
-  out <- pstr_add_reductions(pstr_toy_companies(), pstr_toy_ep_weo(), pstr_toy_weo_2022())
-  new_columns <- setdiff(names(out), names(pstr_toy_companies()))
-  expect_equal(
-    new_columns, c(
-      "EP_categories_id", "EP_group", "weo_product_mapper",
-      "weo_flow_mapper", "publication", "scenario", "region",
-      "category", "unit", "year", "value", "reductions"
+  out <- pstr_add_reductions(companies, ep_weo, weo)
+
+  expected <- c(
+      "company_id",
+      "company_name",
+      "products",
+      "sector",
+      "subsector",
+      "EP_categories_id",
+      "EP_group",
+      "weo_product_mapper",
+      "weo_flow_mapper",
+      "publication",
+      "scenario",
+      "region",
+      "category",
+      "unit",
+      "year",
+      "value",
+      "reductions"
     )
-  )
+  expect_named(out, expected)
 })
 
 test_that("additional columns appear in the output", {
-  out <- pstr_add_reductions(pstr_toy_companies(x = 1), pstr_toy_ep_weo(y = 1), pstr_toy_weo_2022(z = 1))
+  companies <- pstr_toy_companies(x = 1)
+  ep_weo <- pstr_toy_ep_weo(y = 1)
+  weo <- pstr_toy_weo_2022(z = 1)
+
+  out <- pstr_add_reductions(companies, ep_weo, weo)
   expect_true(hasName(out, "x"))
   expect_true(hasName(out, "y"))
   expect_true(hasName(out, "z"))
 })
 
-test_that("preserves typeof() input columns", {
-  out <- pstr_add_reductions(pstr_toy_companies(), pstr_toy_ep_weo(), pstr_toy_weo_2022())
-  x <- unname(purrr::map_chr(pstr_toy_companies(), typeof))
-  y <- unname(purrr::map_chr(out[names(pstr_toy_companies())], typeof))
+test_that("outputs columns of the same type as in `companies`", {
+  companies <- pstr_toy_companies()
+  ep_weo <- pstr_toy_ep_weo()
+  weo <- pstr_toy_weo_2022()
+
+  out <- pstr_add_reductions(companies, ep_weo, weo)
+
+  x <- unname(purrr::map_chr(companies, typeof))
+  y <- unname(purrr::map_chr(out[names(companies)], typeof))
   expect_equal(x, y)
 })
 
-test_that("outputs 0-rows with an empty data frame as input", {
-  out <- pstr_add_reductions(pstr_toy_companies()[FALSE, ], pstr_toy_ep_weo(), pstr_toy_weo_2022())
-  expect_equal(nrow(out), 0L)
+test_that("with 0-row companies outputs 0 rows", {
+  companies <- pstr_toy_companies()
+  ep_weo <- pstr_toy_ep_weo()
+  weo <- pstr_toy_weo_2022()
+  empty <- companies[0L, ]
 
-  out_2 <- pstr_add_reductions(pstr_toy_companies(), pstr_toy_ep_weo()[FALSE, ], pstr_toy_weo_2022())
-  expect_equal(nrow(out), 0L)
+  out <- pstr_add_reductions(empty, ep_weo, weo)
 
-  out_3 <- pstr_add_reductions(pstr_toy_companies(), pstr_toy_ep_weo(), pstr_toy_weo_2022()[FALSE, ])
   expect_equal(nrow(out), 0L)
```

Comment:
Note the typo here. You defined `out_2` and `out_3` but never used them. In any case, the first expectation is enough.

Reply:
Oh right !! Thanks